editor/sidebar/post-status/index.js

```diff
@@ -27,9 +27,14 @@ class PostStatus extends Component {
 	}
 
 	render() {
-		const { status, onUpdateStatus, suggestedFormat } = this.props;
+		const { status, onUpdateStatus, suggestedFormat, post } = this.props;
 		const onToggle = () => {
-			const updatedStatus = status === 'pending' ? 'draft' : 'pending';
+			let updatedStatus;
+			if ( status !== 'pending' ) {
+				updatedStatus = 'pending';
+			} else {
+				updatedStatus = post.status && post.status !== 'pending' ? post.status : 'publish';
```

Comment:
Also, a few minor comments on this function since I found myself staring at it for a few minutes before figuring out what it's for:

- Naming of the function should probably reflect it's toggling related to pending status, e.g. `togglePendingStatus`
- Maybe some inline comments about the `if` behavior paths